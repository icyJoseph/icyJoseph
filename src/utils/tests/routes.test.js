import React from "react";
import { shallow } from "enzyme";
import Routes, {
  AsyncLanding,
  AsyncHacks,
  AsyncBlog,
  AsyncNoMatch,
  AsyncTopMenu,
  AsyncBottomMenu
} from "../../routes";

describe("SnapShot Routes", () => {
  const wrapper = shallow(<Routes />);
  it("renders", () => {
    expect(wrapper).toMatchSnapshot();
  });
});

describe("AsyncLanding", () => {
  const loaded = shallow(<AsyncLanding />);
  it("returns", () => {
    expect(loaded.exists()).toEqual(true);
  });
});

describe("AsyncHacks", () => {
  const loaded = shallow(<AsyncHacks />);
  it("returns", () => {
    expect(loaded.exists()).toEqual(true);
  });
});

describe("AsyncBlog", () => {
  const loaded = shallow(<AsyncBlog />);
  it("returns", () => {
    expect(loaded.exists()).toEqual(true);
  });
});

describe("AsyncNoMatch", () => {
  const loaded = shallow(<AsyncNoMatch />);
  it("returns", () => {
    expect(loaded.exists()).toEqual(true);
  });
});

describe("AsyncTopMenu", () => {
  const loaded = shallow(<AsyncTopMenu />);
  it("returns", () => {
    expect(loaded.exists()).toEqual(true);
  });
});

describe("AsyncBottomMenu", () => {
  const loaded = shallow(<AsyncBottomMenu />);
  it("returns", () => {
    expect(loaded.exists()).toEqual(true);
  });
});
