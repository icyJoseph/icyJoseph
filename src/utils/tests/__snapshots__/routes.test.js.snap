// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SnapShot Routes renders 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <Unknown>
        <LoadableComponent />
        <Switch>
                <Route
                        component={[Function]}
                        path="/blog"
                />
                <Route
                        component={[Function]}
                        path="/hacks"
                />
                <Route
                        component={[Function]}
                        path="/"
                />
                <Route
                        component={[Function]}
                />
        </Switch>
        <LoadableComponent />
</Unknown>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <LoadableComponent />,
          <Switch>
            <Route
                        component={[Function]}
                        path="/blog"
            />
            <Route
                        component={[Function]}
                        path="/hacks"
            />
            <Route
                        component={[Function]}
                        path="/"
            />
            <Route
                        component={[Function]}
            />
</Switch>,
          <LoadableComponent />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <Route
                component={[Function]}
                path="/blog"
/>,
              <Route
                component={[Function]}
                path="/hacks"
/>,
              <Route
                component={[Function]}
                path="/"
/>,
              <Route
                component={[Function]}
/>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "path": "/blog",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "path": "/hacks",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "path": "/",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": Symbol(react.fragment),
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <Unknown>
          <LoadableComponent />
          <Switch>
                    <Route
                              component={[Function]}
                              path="/blog"
                    />
                    <Route
                              component={[Function]}
                              path="/hacks"
                    />
                    <Route
                              component={[Function]}
                              path="/"
                    />
                    <Route
                              component={[Function]}
                    />
          </Switch>
          <LoadableComponent />
</Unknown>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <LoadableComponent />,
            <Switch>
              <Route
                            component={[Function]}
                            path="/blog"
              />
              <Route
                            component={[Function]}
                            path="/hacks"
              />
              <Route
                            component={[Function]}
                            path="/"
              />
              <Route
                            component={[Function]}
              />
</Switch>,
            <LoadableComponent />,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <Route
                  component={[Function]}
                  path="/blog"
/>,
                <Route
                  component={[Function]}
                  path="/hacks"
/>,
                <Route
                  component={[Function]}
                  path="/"
/>,
                <Route
                  component={[Function]}
/>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "component": [Function],
                  "path": "/blog",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "component": [Function],
                  "path": "/hacks",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "component": [Function],
                  "path": "/",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "component": [Function],
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": Symbol(react.fragment),
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
